### ISSUE TITLE
Error on renderingContent in index.d.ts

### ISSUE THREAD

**Issue by sandersn** [algolia/algoliasearch-helper-js#827]
renderingContent doesn't match the expected type inherited from the base type in @algolia/client-search: neither `order` property is optional, and `sortRemainingBy` isn't either.

Here's the compile error I get from a Definitely Typed package that depends on algoliasearch-helper:

```
node_modules/algoliasearch-helper/index.d.ts(1241,5): error TS2416: Property 'renderingContent' in type 'SearchResults<T>' is not assignable to the same property in base type 'Omit<SearchResponse<T>, "facets" | "params">'.
  Type '{ facetOrdering?: { facet?: { order?: string[] | undefined; } | undefined; values?: { [facet: string]: { order?: string[] | undefined; sortRemainingBy?: "hidden" | "alpha" | "count" | undefined; }; } | undefined; } | undefined; } | undefined' is not assignable to type '{ readonly facetOrdering?: { readonly facet?: { readonly order: readonly string[]; } | undefined; readonly values?: { readonly [facet: string]: { readonly order: readonly string[]; readonly sortRemainingBy: "hidden" | "alpha" | "count"; }; } | undefined; } | undefined; } | undefined'.
    Type '{ facetOrdering?: { facet?: { order?: string[] | undefined; } | undefined; values?: { [facet: string]: { order?: string[] | undefined; sortRemainingBy?: "hidden" | "alpha" | "count" | undefined; }; } | undefined; } | undefined; }' is not assignable to type '{ readonly facetOrdering?: { readonly facet?: { readonly order: readonly string[]; } | undefined; readonly values?: { readonly [facet: string]: { readonly order: readonly string[]; readonly sortRemainingBy: "hidden" | "alpha" | "count"; }; } | undefined; } | undefined; }'.
      Types of property 'facetOrdering' are incompatible.
        Type '{ facet?: { order?: string[] | undefined; } | undefined; values?: { [facet: string]: { order?: string[] | undefined; sortRemainingBy?: "hidden" | "alpha" | "count" | undefined; }; } | undefined; } | undefined' is not assignable to type '{ readonly facet?: { readonly order: readonly string[]; } | undefined; readonly values?: { readonly [facet: string]: { readonly order: readonly string[]; readonly sortRemainingBy: "hidden" | "alpha" | "count"; }; } | undefined; } | undefined'.
          Type '{ facet?: { order?: string[] | undefined; } | undefined; values?: { [facet: string]: { order?: string[] | undefined; sortRemainingBy?: "hidden" | "alpha" | "count" | undefined; }; } | undefined; }' is not assignable to type '{ readonly facet?: { readonly order: readonly string[]; } | undefined; readonly values?: { readonly [facet: string]: { readonly order: readonly string[]; readonly sortRemainingBy: "hidden" | "alpha" | "count"; }; } | undefined; }'.
            Types of property 'facet' are incompatible.
              Type '{ order?: string[] | undefined; } | undefined' is not assignable to type '{ readonly order: readonly string[]; } | undefined'.
                Type '{ order?: string[] | undefined; }' is not assignable to type '{ readonly order: readonly string[]; }'.
                  Types of property 'order' are incompatible.
                    Type 'string[] | undefined' is not assignable to type 'readonly string[]'.
                      Type 'undefined' is not assignable to type 'readonly string[]'.
```

Edit: It looks like 41d27f8336f48dd2eaec53afafcceb81a58d0dfb is wrong.

**Comment by Haroenv** [comment]
thanks for opening, it's actually the client which is wrong, and has the keys marked as non-optional. Will fix!

**Comment by Haroenv** [comment]
It's released as @algolia/client-search 4.9.3
